tests/page/page-screenshot.spec.ts

```diff
@@ -48,6 +48,114 @@ it.describe('page screenshot', () => {
+    await page.goto(server.PREFIX + '/rotate-z-shadow-dom.html');
+    const screenshot = await page.screenshot({
+      disableAnimations: true,
+    });
+    for (let i = 0; i < 4; ++i) {
+      const newScreenshot = await page.screenshot({
+        disableAnimations: true,
+      });
+      expect(newScreenshot.equals(screenshot)).toBe(true);
+    }
+  });
+
+  it('should stop animations that happen right before screenshot', async ({ page, server, mode }) => {
+    it.skip(mode !== 'default');
+    await page.goto(server.PREFIX + '/rotate-z.html');
+    // Stop rotating bar.
+    await page.$eval('div', el => el.style.setProperty('animation', 'none'));
+    const buffer1 = await page.screenshot({
+      disableAnimations: true,
+      // Start rotating bar right before screenshot.
+      __testHookBeforeScreenshot: async () => {
+        await page.$eval('div', el => el.style.removeProperty('animation'));
+      },
+    } as any);
+    const buffer2 = await page.screenshot({
+      disableAnimations: true,
+    });
+    expect(buffer1.equals(buffer2)).toBe(true);
+  });
+
+  it('should resume infinite animations', async ({ page, server }) => {
+    await page.goto(server.PREFIX + '/rotate-z.html');
+    await page.screenshot({
+      disableAnimations: true,
+    });
+    const buffer1 = await page.screenshot();
+    const buffer2 = await page.screenshot();
+    expect(buffer1.equals(buffer2)).toBe(false);
+  });
+
+  it('should not capture infinite web animations', async ({ page, server }) => {
+    await page.goto(server.PREFIX + '/web-animation.html');
+    const div = page.locator('div');
+    const screenshot = await div.screenshot({
+      disableAnimations: true,
+    });
+    for (let i = 0; i < 10; ++i) {
+      const newScreenshot = await div.screenshot({
+        disableAnimations: true,
+      });
+      expect(newScreenshot.equals(screenshot)).toBe(true);
+    }
+    // Should resume infinite web animation.
+    const buffer1 = await page.screenshot();
+    const buffer2 = await page.screenshot();
+    expect(buffer1.equals(buffer2)).toBe(false);
+  });
+
+  it('should fire transitionend for finite transitions', async ({ page, server }) => {
+    await page.goto(server.PREFIX + '/css-transition.html');
+    const div = page.locator('div');
+    await div.evaluate(el => {
+      el.addEventListener('transitionend', () => window['__TRANSITION_END'] = true, false);
+    });
+    // Make sure transition is actually running: do two screenshots and compare.
+    expect((await page.screenshot()).equals(await page.screenshot())).toBe(false);
+
+    // Make a screenshot that stops all finite animations.
+    const screenshot1 = await div.screenshot({
+      disableAnimations: true,
+    });
+
+    // Make sure finite transition is not restarted.
+    const screenshot2 = await div.screenshot();
+    expect(screenshot1.equals(screenshot2)).toBe(true);
+
+    expect(await page.evaluate(() => window['__TRANSITION_END'])).toBe(true);
```

Comment:
Let's also check `animation.onfinish`, `animation.oncancel` and `animation.onremove`?

Reply:
done.